app/scripts/migrations/054.js

```diff
@@ -0,0 +1,36 @@
+import { cloneDeep } from 'lodash';
+
+const version = 54;
+
+/**
+ * Migrates preference tokens with 0 decimals typed as 'string' to 'number'
+ */
+export default {
+  version,
+  async migrate(originalVersionedData) {
+    const versionedData = cloneDeep(originalVersionedData);
+    versionedData.meta.version = version;
+    const state = versionedData.data;
+    const newState = transformState(state);
+    versionedData.data = newState;
+    return versionedData;
+  },
+};
+
+function transformState(state) {
+  const newState = state;
+
```

Comment:
A migration of `accountTokens` will also be needed. Check out this PR which is also about to migrate `tokens` and `accountTokens`: https://github.com/MetaMask/metamask-extension/pull/10593/files#diff-0ee44923cc3a183d3803fea2022e1b17f127940f603909b954588cc7c7aa2439R47

And for reference on what those are, see this thread: https://consensys.slack.com/archives/GTQAGKY5V/p1615395521395500?thread_ts=1615332992.380200&cid=GTQAGKY5V

Reply:
Good catch here 👍 I believe having these incorrectly typed in part of `accountTokens` does not produce the UI error in question, but for structural parity they should absolutely be the same.

Reply:
I see that this is addressed now, but for future reference: it does end up indirectly causing the problem, because `tokens` gets derived from `accountTokens` whenever the user switches accounts or networks.

Reply:
@Gudahtt - ah, thank you for pointing this out